### PR TITLE
Upgrade/logic server gui zappy server

### PR DESCRIPTION
**Pull request by AxelF44**
I've upgrade the communication between GUI and Server !

### CODE REVIEW

File: tests/server/tests/communication/communication_test.c
Comment (Thomaltarix) [inline tests/server/tests/communication/communication_test.c:1]: It might be debug

**Review by Thomaltarix — CHANGES_REQUESTED** [review]

**Review by Thomaltarix — APPROVED** [review]

File: gui/src/Engine/Engine.cpp
Comment (mathieurobert1) [inline gui/src/Engine/Engine.cpp:1]: you can delete the comments

File: gui/src/GUIUpdater/GUIUpdater.cpp [hunk 1]
```diff
@@ -411,7 +411,6 @@ void Gui::GUIUpdater::updatePlayerRessourceCollecting(const std::vector<std::str
         for (auto &player : team.getPlayers()) {
             if (player.getId() == args[0]) {
                 player.setState(Gui::Player::PlayerState::COLLECT);
-                _network.get()->sendMessageServer("pin " + std::to_string(player.getId()) + "\n");
                 _gameData.get()->getTile(player.getPosition().first, player.getPosition().second).inventory.removeResource(args[1], 1);
```
Comment (mathieurobert1) [inline gui/src/GUIUpdater/GUIUpdater.cpp:414]:
```suggestion
                player.inventory.addResource(args[1], 1);
                _gameData.get()->getTile(player.getPosition().first, player.getPosition().second).inventory.removeResource(args[1], 1);
```
Can you add this line please, and the same but with **removeResource** in updatePlayerRessourceDropping l.365

**Review by mathieurobert1 — CHANGES_REQUESTED** [review]
Some changes to made please.

**Review by Thomaltarix — APPROVED** [review]

File: gui/src/GUIUpdater/GUIUpdater.cpp [hunk 2]
```diff
@@ -383,6 +383,7 @@ void Gui::GUIUpdater::updatePlayerRessourceDropping(const std::vector<std::strin
         for (auto &player : team.getPlayers()) {
             if (player.getId() == args[0]) {
                 player.setState(Gui::Player::PlayerState::DROP);
+                player.inventory.addResource(args[1], 1);
```
Comment (mathieurobert1) [inline gui/src/GUIUpdater/GUIUpdater.cpp:386]:
```suggestion
                player.inventory.removeResource(args[1], 1);
```

**Review by mathieurobert1 — CHANGES_REQUESTED** [review]
Just one request for the bad explaination in my last request

**Review by Thomaltarix — APPROVED** [review]

**Review by Njord201 — CHANGES_REQUESTED** [review]

**Review by Thomaltarix — APPROVED** [review]
maybe one day

**Review by mathieurobert1 — APPROVED** [review]

**Review by Njord201 — APPROVED** [review]

**Pull request merged**